scripts/triggers-end-to-end-tests/tests.ts

```diff
@@ -274,8 +274,8 @@ describe("storage emulator function triggers", () => {
     // default bucket
     expect(test.storageFinalizedTriggerCount).to.equal(1);
     expect(test.storageV2FinalizedTriggerCount).to.equal(1);
-    expect(test.storageMetadataTriggerCount).to.equal(0);
-    expect(test.storageV2MetadataTriggerCount).to.equal(0);
+    expect(test.storageMetadataTriggerCount).to.equal(1);
```

Comment:
I don't think this is an intended behavior. object upload should only ever emit a finalize trigger, not a metadata trigger

Reply:
same comment through this file

Reply:
Since we do a `metadata.update` in `files.ts` I believe this will now fire for all uploads.

I'm looking at [this code ](https://github.com/firebase/firebase-tools/blob/5564ad1836ca91010e4b99ff78ef70f8f44bdc89/scripts/triggers-end-to-end-tests/functions/index.js#L261) and the [GCS Object Builder docs](https://firebase.google.com/docs/reference/functions/providers_storage.objectbuilder#onmetadataupdate)

Based off of those I think it's expected that these counts should be incremented?

Reply:
Previously the metadata was never updated with whatever was incoming, but now after this change it will be. Since this now counts as a GCS metadata update, I think it makes sense to me that these counts should be incremented?

Reply:
I agree that looking at the code, that's what we would expect to happen but this doesn't conform to the actual spec of GCS storage triggers. Metadata update triggers should only fire on changing the metadata of an *existing* object. 

See https://cloud.google.com/functions/docs/calling/storage#metadata

Reply:
Synced offline, for now I'll add a boolean param to fix this. Left a TODO to refactor the logic in the longer term.